src/Particle/LongRange/StructFact.cpp

```diff
@@ -69,19 +73,32 @@ void StructFact::resize(int ns, int nptcl, int nkpts)
 }
 
 
-void StructFact::UpdateAllPart(ParticleSet& P) { FillRhok(P); }
+void StructFact::UpdateAllPart(const ParticleSet& P)
+{
+  ScopedTimer local(update_all_timer_);
+  FillRhok(P);
+}
+
+void StructFact::mw_UpdateAllPart(const RefVectorWithLeader<StructFact>& sk_list,
+                                  const RefVectorWithLeader<ParticleSet>& p_list)
+{
+  auto& sk_leader = sk_list.getLeader();
+  ScopedTimer local(sk_leader.update_all_timer_);
+  for (int iw = 0; iw < sk_list.size(); iw++)
+    sk_list[iw].FillRhok(p_list[iw]);
+}
 
 
 /** evaluate rok per species, eikr  per particle
  */
-void StructFact::FillRhok(ParticleSet& P)
+void StructFact::FillRhok(const ParticleSet& P)
```

Comment:
Bad method name I'd assume this filled with zeros or something.

What is the actual data dependence? The entire function has to be read in detail to figure it out.  Looks like it could just be a species set.

Reply:
The is function is temporarily kept. It can be deleted, once I implement mw_updateAllParts without calling it.